src/app/AttributeCache.h

```diff
@@ -217,6 +217,37 @@ class AttributeCache : protected ReadClient::Callback
      */
     CHIP_ERROR Get(const ConcreteAttributePath & path, TLV::TLVReader & reader);
 
+    void UpdateClusterSize(ConcreteDataAttributePathWithSize & aCluster)
+    {
+        uint32_t totalSize = 0;
+        ForEachAttribute(aCluster.mEndpointId, aCluster.mClusterId, [this, &totalSize](const ConcreteAttributePath & path) {
```

Comment:
When you fix the logic above to not use the mChangedSet, you'll likely need a new `ForEachEndpoint` iterator as well that you can then nest within that `ForEachCluster` and within that, `ForEachAttribute`.

Reply:
Done,thanks